### PR TITLE
Generic organ

### LINKED ISSUE #2200: Change PLANT metabolic nitrogen retranslocation and reallocation calculations
Continuing on #2138 and #2141, in the generic organ, metabolic component doesn't contribute to nitrogen retranslocation and reallocation which should include if the metabolic component is not zero.

### PR DESCRIPTION
**Pull request by HamishBrownPFR**
resolves #2200 

Added some code to reallocate and retranslocate N from generic organ

### CODE REVIEW

**Comment by HamishBrownPFR** [comment]
@byzheng I have added some code to deal with metabolic N in generic organ.  Take a look and let me know if you see any problems

**Comment by byzheng** [comment]
@HamishBrownPFR. It is one solution that N proportionally reduces from storage and metabolic. However, for my understanding (correct me if it is wrong), retransllocation should firstly reduce the storage N, and the metabolic N.

**Comment by HamishBrownPFR** [comment]
@byzheng Good point, Ill have another go

**Comment by HamishBrownPFR** [comment]
@byzheng this may be better

**Comment by HamishBrownPFR** [comment]
diffs in Chicory all seem trivial, @rcichota can you please check and confirm.  Diffs in patch test are due to some variables now being assigned a number when they were missing.  I am not sure why this test is there, it doesn't seem to have any observed data.

Can merge if @rcichota agrees effect on chicory is trivial.

**Comment by rcichota** [comment]
Yes this can be merged. The diffs are greater in those variables that were already quite poorly predicted. After talking to Hamish and checking what is being changed, I guess that the diffs arise from chicory using re-translocation and reallocation more than other species.  I will raise an issue so that the parameters in chicory can be revised, I have also just noticed that some data is off by one day, which caused some of the poor statistics... this may be true also for plantain...

File: Models/Plant/Organs/GenericOrgan.cs [hunk 1]
```diff
@@ -389,15 +388,19 @@ public virtual void SetNitrogenAllocation(BiomassAllocationType nitrogen)
...
+            double StorageNRetranslocation = Math.Min(nitrogen.Retranslocation,startLive.StorageN * (1 - senescenceRate.Value()) * nRetranslocationFactor.Value());
+            Live.StorageN -= StorageNRetranslocation;
+            Live.MetabolicN -= (nitrogen.Retranslocation - StorageNRetranslocation);
             Allocated.StorageN -= nitrogen.Retranslocation;
 
             // Reallocation
-            if (MathUtilities.IsGreaterThan(nitrogen.Reallocation, startLive.StorageN))
-                throw new Exception("N reallocation exceeds non structural nitrogen in organ: " + Name);
-            Live.StorageN -= nitrogen.Reallocation;
+            if (MathUtilities.IsGreaterThan(nitrogen.Reallocation, startLive.StorageN + startLive.MetabolicN))
```
Comment (byzheng) [inline Models/Plant/Organs/GenericOrgan.cs]: Should like this?
```csharp
if (MathUtilities.IsGreaterThan(nitrogen.Reallocation, (startLive.StorageN + startLive.MetabolicN) * (1- * senescenceRate.Value()))
```

**Comment by byzheng** [comment]
@HamishBrownPFR Look reasonable to me now.

**Comment by HamishBrownPFR** [comment]
Thanks @byzheng and @rcichota.  @hol353 this can be merged if you are happy with it

**Pull request merged**